### PR TITLE
NCGenerics: restrict conditional Copyable reqs

### PR DESCRIPTION
**Pull request by kavon**
It doesn't really make sense for a conditional conformance requirement for `Copyable` to depend on any other requirement other than other `Copyable` conformance requirements.

resolves rdar://124967739

### CODE REVIEW

**Comment by kavon** [comment]
@swift-ci smoke test

File: lib/Sema/TypeCheckInvertible.cpp [hunk 1]
```diff
@@ -140,10 +140,45 @@ static void checkInvertibleConformanceCommon(DeclContext *dc,
...
+      auto condReqs = normalConf->getConditionalRequirements();
+      hasUnconditionalConformance = condReqs.empty();
+      auto *thisProto = normalConf->getProtocol();
+
+      // Ensure that conditional conformance to an invertible protocol IP only
+      // depends conformance requirements involving IP.
+      //
+      // In theory, it could depend on any invertible protocol, but it may be
+      // confusing if we permitted that and this simplifies the model a bit.
+      for (auto req : condReqs) {
```
Comment (slavapestov) [inline lib/Sema/TypeCheckInvertible.cpp:155]: Can you also ban requirements where the first type is a DependentMemberType?

File: lib/Sema/TypeCheckInvertible.cpp [hunk 2]
```diff
@@ -140,10 +140,45 @@ static void checkInvertibleConformanceCommon(DeclContext *dc,
...
+            break; // permitted, don't fill-in.
+        LLVM_FALLTHROUGH;
+        case RequirementKind::Superclass:
+        case RequirementKind::SameType:
+        case RequirementKind::SameShape:
+          illegalSecondTypeStr = req.getSecondType().getString();
+          break;
+        }
+
+        if (illegalSecondTypeStr)
```
Comment (slavapestov) [inline lib/Sema/TypeCheckInvertible.cpp:174]: Nitpick: Add { } around multi-line if body

**Comment by kavon** [comment]
@swift-ci smoke test

**Comment by kavon** [comment]
@swift-ci smoke test

**Comment by kavon** [comment]
@swift-ci smoke test

**Comment by kavon** [comment]
@swift-ci smoke test linux

**Comment by kavon** [comment]
@swift-ci smoke test

**Pull request merged**